packages/web3torrent/src/components/torrent-info/channels-list/ChannelsList.tsx

```diff
@@ -23,53 +24,62 @@ class ChannelsList extends React.Component<UploadInfoProps> {
   channelIdToTableRow(
     channelId: string,
     channels: Dictionary<ChannelState>,
-    wires: PaidStreamingWire[],
+    torrent: Torrent,
     participantType: 'payer' | 'beneficiary'
   ) {
     let channelButton;
     const channel = channels[channelId];
 
+    const wire = torrent.wires.find(
+      wire =>
+        wire.paidStreamingExtension.peerChannelId === channelId ||
+        wire.paidStreamingExtension.pseChannelId === channelId
+    );
     if (channel.status === 'closing') {
       channelButton = <button disabled>Closing ...</button>;
     } else if (channel.status === 'closed') {
       channelButton = <button disabled>Closed</button>;
+    } else if (channel.status === 'challenging') {
+      channelButton = <button disabled>Challenging</button>;
     } else {
-      channelButton = <button disabled>Running</button>;
+      channelButton = getPeerStatus(torrent, wire) ? (
+        <button disabled>Running</button>
+      ) : (
+        <button
+          className="button-alt"
+          onClick={_ => this.context.paymentChannelClient.challengeChannel(channelId)}
+        >
+          Challenge Channel
+        </button>
+      );
     }
 
-    const wire = wires.find(
-      wire =>
-        wire.paidStreamingExtension.peerChannelId === channelId ||
-        wire.paidStreamingExtension.pseChannelId === channelId
-    );
-
-    let transferred: string;
+    let dataTransferred: string;
     const peerAccount = channel[participantType];
     if (wire) {
-      transferred =
-        participantType === 'beneficiary' ? prettier(wire.uploaded) : prettier(wire.downloaded);
+      dataTransferred =
+        participantType === 'beneficiary'
+          ? prettier(wire.uploaded) + ` up`
+          : prettier(wire.downloaded) + ` down`;
     } else {
       // Use the beneficiery balance as an approximate of the file size, when wire is dropped.
-      transferred = prettyPrintBytes(utils.bigNumberify(channel.beneficiaryBalance));
+      dataTransferred =
+        prettyPrintBytes(utils.bigNumberify(channel.beneficiaryBalance)) +
+        (participantType === 'beneficiary' ? ` up` : ` down`);
     }
 
+    const weiTransferred = prettyPrintWei(utils.bigNumberify(channel.beneficiaryBalance));
+
     return (
       <tr className="peerInfo" key={channelId}>
         <td>{channelButton}</td>
         <td className="channel-id">{channelId}</td>
         <td className="peer-id">{peerAccount}</td>
-        <td className="uploaded">
-          {transferred}
-          {participantType === 'beneficiary' ? ` up` : ` down`}
-        </td>
+        <td className="uploaded">{dataTransferred}</td>
         {participantType === 'beneficiary' ? (
-          <td className="earned">
-            {prettyPrintWei(utils.bigNumberify(channel.beneficiaryBalance))}
-          </td>
+          <td className="earned">{weiTransferred}</td>
         ) : (
-          <td className="paid">
```

Comment:
I made a little refactor here, to improve (IMO) readability, if someone disagrees, I can change it back